src/main/java/com/couchbase/connect/kafka/handler/source/SourceHandler.java

```diff
@@ -30,6 +33,18 @@ public interface SourceHandler {
   default void init(Map<String, String> configProperties) {
   }
 
+  /**
+   * Function called as acknowledgement of a SourceRecord being published to a Kafka topic, propagated from Kafka Connect SourceTask.commitRecord().
+   * SourceHandlers are not required to implement this function, by default it is a no-op.
+   * 
+   * @param sourceRecord {@link SourceRecord} that was successfully sent via the producer or filtered by a transformation
+   * @param recordMetadata {@link RecordMetadata} record metadata returned from the broker, or null if the record was filtered
+   * @param isBlackHole boolean indicating whether the topic is a black hole topic (ignored events) [true] or regular topic [false]
+   */
+  default void recordCommited(SourceRecord sourceRecord, RecordMetadata recordMetadata, boolean isBlackHole) {
```

Comment:
Since the metadata can be null, can you please annotate the `RecordMetadata` parameter type as `@Nullable`? The annotation from package `org.jspecify.annotations`.

    @Nullable RecordMetadata recordMetadata

More info about JSpecify, if you're interested: https://jspecify.dev